app/screens/map-screen/map-screen.tsx

```diff
@@ -95,168 +120,120 @@ export const MapScreen: React.FC<Props> = ({ navigation }) => {
   }
 
-  const maps = data?.businessMapMarkers ?? []
+  // On screen load, check (NOT request) if location permissions are given
+  React.useEffect(() => {
+    check(LOCATION_PERMISSION).then((status) => {
+      console.log("Initial check: ", status)
+      setPermissionsStatus(status)
+      if (status === RESULTS.GRANTED) {
+        getUserRegion(async (region) => {
+          if (region) {
+            setInitialLocation(region)
+          } else {
+            setInitializing(false)
+          }
+        })
+      } else {
+        setInitializing(false)
+      }
+    })
+  }, [])
+
+  const alertOnLocationError = React.useCallback(() => {
+    Alert.alert(LL.common.error(), LL.MapScreen.error())
+  }, [LL])
+
+  React.useEffect(() => {
+    if (lastCoordsError) {
+      setInitializing(false)
+      setInitialLocation(EL_ZONTE_COORDS)
+      alertOnLocationError()
+    }
+  }, [lastCoordsError, alertOnLocationError])
 
-  // if getting location was denied and device's country code has been found (or defaulted)
-  // this is used to finalize the initial location shown on the Map
+  // Flow when location permissions are denied
   React.useEffect(() => {
-    if (countryCode && wasLocationDenied && !loading) {
-      // JSON 'hashmap' with every countrys' code listed with their lat and lng
-      const countryCodesToCoords: {
-        data: Record<CountryCode, { lat: number; lng: number }>
-      } = JSON.parse(JSON.stringify(countryCodes))
-      const countryCoords: { lat: number; lng: number } =
-        countryCodesToCoords.data[countryCode]
-      if (countryCoords) {
+    if (
+      countryCode &&
+      lastCoordsData &&
+      !isInitializing &&
+      !loading &&
+      !initialLocation
+    ) {
+      // User has used map before, so we use their last viewed coords
+      if (lastCoordsData.latLng?.lat && lastCoordsData.latLng?.lng) {
         const region: Region = {
-          latitude: countryCoords.lat,
-          longitude: countryCoords.lng,
-          latitudeDelta: LATITUDE_DELTA,
+          latitude: lastCoordsData.latLng.lat,
+          longitude: lastCoordsData.latLng.lng,
+          latitudeDelta: 5,
           longitudeDelta: LONGITUDE_DELTA,
         }
-        setUserLocation(region)
-      }
-      setIsLoadingLocation(false)
-    }
-  }, [wasLocationDenied, countryCode, loading, setIsLoadingLocation, setUserLocation])
-
-  const getUserRegion = (callback: (region?: Region) => void) => {
-    try {
-      Geolocation.getCurrentPosition(
-        (data: GeolocationPosition) => {
-          if (data) {
-            const region: Region = {
-              latitude: data.coords.latitude,
-              longitude: data.coords.longitude,
-              latitudeDelta: 0.02,
-              longitudeDelta: 0.02,
-            }
-            callback(region)
+        setInitialLocation(region)
+        // User is using maps for the first time, so we center on the center of their IP's country
```

Comment:
is this comment for the else branch below? if that is the case, I would move this line below the else, to be consistent with [comment line 163](https://github.com/GaloyMoney/galoy-mobile/pull/2999/files#diff-2840b3b05c86d0fba4f9261d1ba0aaeca6798d1ad37be559b44495d847710f04R163) where the comment is within the block, not before it